### ISSUE TITLE
Can not go get  github.com/pingcap/tidb-operator

### ISSUE THREAD

**Issue by Ehco1996** [pingcap/tidb-operator#4009]
## Question

Before asking a question, make sure you have:

- Reviewed relevant Kubernetes information: Google your error messages and look at K8s docs.
- Searched open and closed [GitHub issues](https://github.com/pingcap/tidb-operator/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Read the documentation:
  - [TiDB Operator Readme](https://github.com/pingcap/tidb-operator)
  - [TiDB Operator Doc](https://github.com/pingcap/tidb-operator/tree/master/docs)

---

as title

`go get -v  -u github.com/pingcap/tidb-operator@v1.2.0-rc.1`

get
```
go get: github.com/pingcap/tidb-operator@v1.0.0-beta.1-p2 updating to
	github.com/pingcap/tidb-operator@v1.2.0-rc.1 requires
	github.com/uber-go/atomic@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

My question is why tidb-operator use  this verison `v0.0.0-00010101000000-000000000000` of `uber-go/atomic` ?

can we just update the dep ?

**Comment by DanielZhangQD** [comment]
@Ehco1996 There are some complicated dependencies in TiDB Operator now.
We will improve them later.
BTW, would you like to help resolve this issue?

**Comment by Ehco1996** [comment]
> @Ehco1996 There are some complicated dependencies in TiDB Operator now.
> We will improve them later.
> BTW, would you like to help resolve this issue?

i can have a try 

but i found many dependencies with replace beacuse [upstream k8s's go mod](https://github.com/kubernetes/kubernetes/blob/master/go.mod) use replace and i dont't know how to handle this

could you give any advice?

**Comment by DanielZhangQD** [comment]
For the dependencies caused by `kubernetes` https://github.com/pingcap/tidb-operator/blob/master/go.mod#L95, please ignore them, we will try to remove the dependency with the `kubernetes` directly later.
You can help check if any other dependencies can be optimized.
Thanks!

**Comment by DanielZhangQD** [comment]
@Ehco1996 We have separated the packages `pkg/apis` and `pkg/client` into submodules in https://github.com/pingcap/tidb-operator/pull/4134, could you please check this again?

**Comment by Ehco1996** [comment]
> 

I'll give it a try~~

**Comment by Ehco1996** [comment]
@DanielZhangQD 

`pkg/apis` is works ! Great job!

```
go get -u -v github.com/pingcap/tidb-operator/pkg/apis
go: downloading github.com/pingcap/tidb-operator/pkg/apis v0.0.0-20211213052434-dc276dee3d3c
go get: added github.com/pingcap/tidb-operator/pkg/apis v0.0.0-20211213052434-dc276dee3d3c
```

but for `pkg/client` is still can't import 
```
play on  master via 🐹 v1.17.4 at ☸️  k3s (proxy) took 2s
➜ go get -u -v github.com/pingcap/tidb-operator/pkg/client@master
go: downloading github.com/pingcap/tidb-operator v1.2.0-rc.2.0.20211213052434-dc276dee3d3c
go get: github.com/pingcap/tidb-operator/pkg/client@v0.0.0-20211213052434-dc276dee3d3c requires
	github.com/pingcap/tidb-operator/pkg/apis@v0.0.0: reading github.com/pingcap/tidb-operator/pkg/apis/pkg/apis/go.mod at revision pkg/apis/v0.0.0: unknown revision pkg/apis/v0.0.0
```

**Comment by KanShiori** [comment]
> @DanielZhangQD
> 
> `pkg/apis` is works ! Great job!
> 
> ```
> go get -u -v github.com/pingcap/tidb-operator/pkg/apis
> go: downloading github.com/pingcap/tidb-operator/pkg/apis v0.0.0-20211213052434-dc276dee3d3c
> go get: added github.com/pingcap/tidb-operator/pkg/apis v0.0.0-20211213052434-dc276dee3d3c
> ```
> 
> but for `pkg/client` is still can't import
> 
> ```
> play on  master via 🐹 v1.17.4 at ☸️  k3s (proxy) took 2s
> ➜ go get -u -v github.com/pingcap/tidb-operator/pkg/client@master
> go: downloading github.com/pingcap/tidb-operator v1.2.0-rc.2.0.20211213052434-dc276dee3d3c
> go get: github.com/pingcap/tidb-operator/pkg/client@v0.0.0-20211213052434-dc276dee3d3c requires
> 	github.com/pingcap/tidb-operator/pkg/apis@v0.0.0: reading github.com/pingcap/tidb-operator/pkg/apis/pkg/apis/go.mod at revision pkg/apis/v0.0.0: unknown revision pkg/apis/v0.0.0
> ```

Because pkg api don't have a release version, it will be fixed after release operator v1.3. Ref issue https://github.com/pingcap/tidb-operator/issues/4138

**Comment by Ehco1996** [comment]
part Duplicate of https://github.com/pingcap/tidb-operator/issues/4138, so i think i can close this issue now
